workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.test.ts

```diff
@@ -77,4 +108,160 @@ describe('WBCGroupManagemntPlugin', () => {
       );
     });
   });
+
+  describe('createIdentityPermissions', () => {
+    let sampleGroupId: string;
+    let sampleGroupType: IdentityType;
+
+    let sampleAction: Action;
+    let sampleEffect: Effect;
+    let sampleSubjectType: string;
+    let sampleSubjectId: string;
+    let sampleConditions: Record<string, JSONValue>;
+    let sampleFields: string[];
+    let sampleDescription: string;
+
+    let mockIdentityPermission: IdentityPermission;
+
+    let mockIdentityPermissions: IdentityPermission[];
+
+    let actor: object;
+    let source: object;
+    let action: string;
+
+    beforeAll(() => {
+      jest.spyOn(auditService, 'write');
```

Comment:
What is this for? Usually youre setting the return value as a spy variable or using it to mock implementation

Reply:
This is used to track method calls

Reply:
how, usually you do something like `const spy = jest.spyOn(...); expect(spy).toHaveBeenCalled()` but I dont see that here

Reply:
spyOn attaches a listener to the object's method and tracks call to that method

Reply:
I would suggest creating a spy variable and using that variable to check that the function was called with certain params. Less confusing and follows Jest's docs on how to implement spies